### PR TITLE
Add npmignore to cut down on package size

### PR DESCRIPTION
**Pull request by amsardesai**
If we add a `.npmignore` file, we can cut down on the number of files downloaded when users who use this package do `npm install`.

### CODE REVIEW

**Comment by vramana** [comment]
@amsardesai Please merge the commits into one.

**Comment by amsardesai** [comment]
@vramana done

**Comment by vramana** [comment]
@amsardesai Thanks for your contribution!!

File: .npmignore [hunk 1]
```diff
@@ -1 +1,17 @@
...
+bower_components
+.sass-cache
+build
+demos
+demos1
+TODO.md
+test
+testlib
+docs
+dist
```
Comment (akiran) [inline .npmignore]: It is better to keep the dist folder.
Some people may prefer to add those files in script tags instead of using browserify/webpack

Comment (amsardesai) [inline .npmignore]: Hmm, good point. I'll make another PR.

**Pull request merged**